common/server_player.cpp

```diff
@@ -1522,7 +1522,14 @@ Response::ResponseCode Server_Player::cmdRevealCards(const Command_RevealCards &
         return Response::RespNameNotFound;
     
     QList<Server_Card *> cardsToReveal;
-    if (!cmd.has_card_id())
+    if (cmd.top_cards() != -1) {
+        for (int i = 0; i < cmd.top_cards(); i++) {
+            Server_Card *card = zone->getCard(i);
+            if (!card)
+                return Response::RespNameNotFound;
+            cardsToReveal.append(card);
```

Comment:
Derp you're right

On Sat, May 9, 2015, 5:19 AM ctrlaltca <notifications@github.com> wrote:

> In common/server_player.cpp
> <https://github.com/Cockatrice/Cockatrice/pull/1051#discussion_r29991328>:
>
> > @@ -1522,7 +1522,14 @@ Response::ResponseCode Server_Player::cmdRevealCards(const Command_RevealCards &
> >          return Response::RespNameNotFound;
> >
> >      QList<Server_Card *> cardsToReveal;
> > -    if (!cmd.has_card_id())
> > +    if (cmd.top_cards() != -1) {
> > +        for (int i = 0; i < cmd.top_cards(); i++) {
> > +            Server_Card *card = zone->getCard(i);
> > +            if (!card)
> > +                return Response::RespNameNotFound;
> > +            cardsToReveal.append(card);
>
> As far as i understand the code, no new object gets created here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Cockatrice/Cockatrice/pull/1051/files#r29991328>.
>